Source/Language/IReturns.cs

```diff
@@ -67,6 +68,26 @@ public partial interface IReturns<TMock, TResult> : IFluentInterface
 		IReturnsResult<TMock> Returns(TResult value);
 
 		/// <summary>
+		/// Specifies a function that will calculate the value to return from an open generic method.
+		/// </summary>
+		/// <param name="valueFunction">
+		/// The function that will calculate the return value.
+		/// IPublicCallContext contains the closed generic method as well as the arguments passed in.
+		/// </param>
+		/// <example group="returns">
+		/// Return a calculated value when the method is called:
+		/// <code>
+		/// mock.Setup(x => x.Execute&lt;It.IsAnyType>())
+		///     .Returns(context => returnValues[0]);
+		/// </code>
+		/// The lambda expression to retrieve the return value is lazy-executed, 
+		/// meaning that its value may change depending on the moment the method 
+		/// is executed and the value the <c>returnValues</c> array has at 
+		/// that moment.
+		/// </example>
+		IReturnsResult<TMock> Returns(Func<IPublicCallContext, object> valueFunction);
```

Comment:
The following now does not compile, as a `.Setup()' no that is not 'It.AnyType return type in the expression' does not contain the 'Returns(Func<T,object>)' overload anymore:

`new Mock<IBar>().Setup(x => x.Foo).Returns(context => 2);`

I did have a question regarding the documentation for the following:

```
// TODO: Find out about this (.xdoc?)
/// <include file='Mock.Generic.xdoc' path='docs/doc[@for="Mock{T}.Setup{}"]/*'/>
public ISetupAnyType<T, It.AnyType> Setup(Expression<Func<T, It.AnyType>> expression)
{
	return Mock.Setup(this, expression, null);
}
```

Wasn't sure what to put as the comment/documentation, and not sure how the current documentation works, any pointers would be great!

Currently to setup a method with an Open Generic return type that has a where clause requires a direct case (so that the generic `.Setup<TResult>(Expression<Func<TMock,TResult>>)` overload does not kick in. See updated test:

```
// Have to cast to (Expression<Func<IHasOpenGeneric, It.AnyType>>)
mock.Setup((Expression<Func<IHasOpenGeneric, It.AnyType>>)(x => x.GetWhere<ItAnySomeBaseClass, ItAnySomeBaseClass, ItAnySomeBaseClass>(It.IsAny<ItAnySomeBaseClass>())))
	.Returns(context =>
	{
		callContext = context;
		callCount++;
		return new SomeImplementationClass();
	});
```

I will be taking a look at changing SubClassing via It.AnySubTypeOf<T>, see if I get anywhere with that (as part of your `Source/It.cs` comment)